### PR TITLE
fix crash attachment

### PR DESCRIPTION
**Pull request by namhihi237**
### Details

This PR is to try to fix regression from [PR](https://github.com/Expensify/App/pull/33586)
### Fixed Issues

$ https://github.com/Expensify/App/issues/34016
$ https://github.com/Expensify/App/issues/33985
PROPOSAL:

### Tests
Case 1:
1. Go to workspace chat
2. Click + > Request money > Distance
3. Create a distance request and verify that no console error 404 after creating a distance request

Case 2:
1. Go to any room or profile details page
2. Open the room or profile picture verify that the app does not crash and show the picture

- [x] Verify that no errors appear in the JS console

### Offline tests
The same test

### QA Steps
The same test

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/39086067/5e6304bf-9864-4b80-ae62-73accadd90e3

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/39086067/39cd92b4-e570-42c7-a1e8-6ce7d41f07b0

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/39086067/91422ed5-dcf9-4cd4-9c7d-af2742be0961

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/39086067/72b074bd-057f-40ae-b950-07fbb31d24d8

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/39086067/925dc2ed-78dd-4f96-96e6-414da10b0c60

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/39086067/ce336317-8f5f-444b-a9fe-f690bdbb2c10

</details>

### CODE REVIEW

**Review by 0xmiroslav — APPROVED** [review]
Code looks good. Testing

File: src/libs/ReceiptUtils.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ function getThumbnailAndImageURIs(transaction: Transaction, receiptPath: string
         image = ReceiptSVG;
     }
 
-    const isLocalFile = typeof path === 'number' || path.startsWith('blob:') || path.startsWith('file:');
+    const isLocalFile = typeof path === 'number' || path.startsWith('blob:') || path.startsWith('file:') || path === ReceiptGeneric;
```
[the remark this answers is not in the document]
Comment (namhihi237) [reply src/libs/ReceiptUtils.ts:69]: I think `ImageSourcePropType` from `import type {ImageSourcePropType} from 'react-native';`

[the remark this answers is not in the document]
Comment (namhihi237) [reply src/libs/ReceiptUtils.ts:69]: Ah it's string.

**Comment by namhihi237** [comment]
Seem it does not relative this PR

**Comment by namhihi237** [comment]
I test on production and this issue still happen

**Review by 0xmiroslav — APPROVED** [review]
@thienlnam this needs CP'ed

**Comment by sobitneupane** [comment]
@namhihi237 Is [this issue](https://github.com/Expensify/App/issues/34047) also a regression? Will it be handled by the PR?

**Comment by namhihi237** [comment]
I will tested this

**Comment by namhihi237** [comment]
I can not reproduce on the main, anyone can reproduce? I test it to work on main and this PR.

**Comment by namhihi237** [comment]
PR will fix this issue https://github.com/Expensify/App/issues/34049

**Comment by sobitneupane** [comment]
@namhihi237 Let's make sure https://github.com/Expensify/App/issues/34049 issue is also handled by this PR.

**Comment by namhihi237** [comment]
Testing on it

**Comment by namhihi237** [comment]
Tested and was cover

**Comment by sobitneupane** [comment]
Thank you @namhihi237 for quick response. Let's keep an eye open for probable deploy blockers.

File: src/libs/ReceiptUtils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (thienlnam) [reply src/libs/ReceiptUtils.ts:69]: Hmm doesn't this apply to all paths that have a local file then?

[the remark this answers is not in the document]
Comment (namhihi237) [reply src/libs/ReceiptUtils.ts:69]: Yes, I think if we check isLocalFile we should use this, currently I see we build the isLocalFile only in this function `getThumbnailAndImageURIs`

[the remark this answers is not in the document]
Comment (thienlnam) [reply src/libs/ReceiptUtils.ts:69]: Can we apply a more hollistic approach to isLocalFile though? I imagine in the future we'll have more web assets we pass into getThumbnailAndImageURIs and ideally we don't just add an if check for each path. Maybe we can check if it starts with '/'? What do you think

[the remark this answers is not in the document]
Comment (namhihi237) [reply src/libs/ReceiptUtils.ts:69]: Hmm, I think it is good, we can modify it like that to cover all we assets' path:
```js
const isLocalFile = ... path.startsWith('/');
```

**Comment by namhihi237** [comment]
@thienlnam PR updated

**Comment by namhihi237** [comment]
https://github.com/Expensify/App/assets/39086067/ddb5a473-8615-41aa-8eef-775a5248ec62


https://github.com/Expensify/App/assets/39086067/d87766d8-2f98-402d-a7c1-37f877b14292

**Review by thienlnam — APPROVED** [review]
Lovely, thanks!

File: src/libs/ReceiptUtils.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ function getThumbnailAndImageURIs(transaction: Transaction, receiptPath: string
         image = ReceiptSVG;
     }
 
-    const isLocalFile = typeof path === 'number' || path.startsWith('blob:') || path.startsWith('file:');
+    const isLocalFile = typeof path === 'number' || path.startsWith('blob:') || path.startsWith('file:') || path.startsWith('/');
```
Comment (shubham1206agra) [inline src/libs/ReceiptUtils.ts:69]: This PR created a problem with eReceipts that have pending waypoints. Issue https://github.com/Expensify/App/issues/34287

**Pull request merged**